k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java

```diff
@@ -258,9 +259,13 @@ private void showAdditionalHeaders() {
 
     public void populate(final Message message, final Account account) {
         final Contacts contacts = K9.showContactName() ? mContacts : null;
-        final CharSequence from = MessageHelper.toFriendly(message.getFrom(), contacts);
+        CharSequence from = MessageHelper.toFriendly(message.getFrom(), contacts);
         final CharSequence to = MessageHelper.toFriendly(message.getRecipients(Message.RecipientType.TO), contacts);
         final CharSequence cc = MessageHelper.toFriendly(message.getRecipients(Message.RecipientType.CC), contacts);
+        if(showSender(message)) {
+            from = MessageHelper.toFriendly(message.getSender(), contacts) + " " +
```

Comment:
The translation-friendly way of doing this is having a format string with placeholders for the names, e.g. "%1$s on behalf of %2$s".